controller/contour/deployment_test.go

```diff
@@ -0,0 +1,142 @@
+package contour
+
+import (
+	"testing"
+
+	operatorv1alpha1 "github.com/projectcontour/contour-operator/api/v1alpha1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO [danehans]: Add integration tests to test reconcilitation
+// after https://github.com/projectcontour/contour-operator/pull/29 merges.
+
+var (
+	testName = "test-deploy"
+	testNs   = testName + "-ns"
+)
+
+func checkDeploymentHasEnvVar(t *testing.T, deploy *appsv1.Deployment, name string) {
+	t.Helper()
+
+	for _, envVar := range deploy.Spec.Template.Spec.Containers[0].Env {
+		if envVar.Name == name {
+			break
+		}
+	}
+}
+
+func checkDeploymentHasContainer(t *testing.T, deploy *appsv1.Deployment, name string, expect bool) *corev1.Container {
+	t.Helper()
+
+	for _, container := range deploy.Spec.Template.Spec.Containers {
+		if container.Name == name {
+			if expect {
+				return &container
+			}
+			t.Errorf("deployment has unexpected %q container", name)
+		}
+	}
+	if expect {
+		t.Errorf("deployment has no %q container", name)
+	}
+	return nil
+}
+
+func checkContainerHasImage(t *testing.T, container *corev1.Container, image string) {
+	t.Helper()
+
+	if container.Image == image {
+		return
+	} else {
+		t.Errorf("deployment has invalid image %q", container.Image)
+	}
+}
+
+func TestDesiredDeployment(t *testing.T) {
+	ctr := &operatorv1alpha1.Contour{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testName,
+			Namespace: testNs,
+		},
+	}
+	deploy, err := desiredDeployment(ctr)
+	if err != nil {
+		t.Errorf("invalid deployment: %v", err)
+	}
+
+	container := checkDeploymentHasContainer(t, deploy, deployContainerName, true)
```

Comment:
Need a nil check, or need to change `checkDeploymentHasContainer` to use `t.Fatalf` instead of `t.Errorf`.